services/requester/keystore/key_store.go

```diff
@@ -106,3 +132,50 @@ func (k *KeyStore) setLockMetadata(
 	defer k.keyMu.Unlock()
 	k.usedKeys[txID] = key
 }
+
+func (k *KeyStore) processLockedKeys(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			close(k.done)
+			return
+		case blockID := <-k.blockChan:
+			// TODO: If calling `k.client.GetTransactionResultsByBlockID` for each
```

Comment:
This is something we need to consider. If calling `GetTransactionResultsByBlockID` for each new block, is a bit aggressive for ANs, we might consider the approach with the available keys ratio.